.github/workflows/test-and-release-svm-contract.yml

```diff
@@ -1,30 +1,41 @@
-name: Release SVM Program
+name: Test and release SVM contracts
 
 on:
+  pull_request:
+    paths:
+      - contracts/svm/**
   push:
     branches:
       - main
     paths:
       - contracts/svm/**
 
+env:
+  CARGO_TERM_COLOR: always
+
 jobs:
   build:
     runs-on: ubuntu-latest
-
+    defaults:
+      run:
+        working-directory: ./contracts/svm
     steps:
       - uses: actions/checkout@v2
       - name: Install Solana Verify CLI
         run: |
           cargo install solana-verify --git https://github.com/Ellipsis-Labs/solana-verifiable-build --rev 121568e
       - name: Build
-        working-directory: ./contracts/svm
         run: solana-verify build
+      - name: Run tests
+        env:
+          SBF_OUT_DIR: ${{ github.workspace }}/contracts/svm/target/deploy
+        run: cargo test --target-dir ./target-tests
```

Comment:
we use `target-tests/` because `target/` becomes readonly after `solana-verify build` for some reasons I don't fully understand... but it's convenient because it means the artifact doesn't get updated at this step

Reply:
also avoid `cargo-test-sbf` because it rebuilds the binary